sdk/sdk-api/src/main/java/mil/emp3/api/MilStdSymbol.java

```diff
@@ -795,25 +796,37 @@ public float getNumericModifier(IGeoMilSymbol.Modifier eModifier, int iIndex) {
 
     /**
      * This method retrieves the basic symbol code with no modifiers.
-     * @return
+     * @throws IllegalStateException if called before symbol code is set
+     * @return basic symbol code
      */
     public String getBasicSymbol() {
+        if (getSymbolCode() == null) {
+            throw new IllegalStateException("Can't call this method when symbol code is null");
+        }
         return SymbolUtilities.getBasicSymbolID(this.getSymbolCode());
     }
 
     /**
      * This method trues true if the symbol code is a tactical graphic. it returns false otherwise.
-     * @return
+     * @throws IllegalStateException if called before symbol code is set
+     * @return true if symbol is a tactical graphic
      */
     public boolean isTacticalGraphic() {
+        if (getSymbolCode() == null) {
+            throw new IllegalStateException("Can't call this method when symbol code is null");
+        }
```

Comment:
Instead of repeating this coder everywhere, why not create a method checkForNullSymbolCode() that will throw the exception. Then simply invoke that method and repetitive if ... clause and throw exception can be elimitanted.